DemoApp/Sources/Components/DemoAppViewFactory.swift

```diff
@@ -98,7 +98,7 @@ final class DemoAppViewFactory: ViewFactory {
                 onChangeTrackVisibility: onChangeTrackVisibility
             )
             .snapshot(trigger: snapshotTrigger) { [weak viewModel] snapshot in
```

Comment:
do we still need it here?

Reply:
If we don't add it as weak in the external closure the model will be captured strongly. It's not producing a warning but it's not correct to strongly capture a reference type in a closure.